manifests/params.pp

```diff
@@ -1,34 +1,32 @@
 class bdii::params {
- 
-  $log_level            = hiera("bdii::params::log_level","DEBUG")
-  $port                 = hiera("bdii::params::port","2170")
-  $user                 = hiera("bdii::params::user","ldap")
-  $slapdconf            = hiera("bdii::params::slapdconf","/etc/bdii/bdii-slapd.conf")
-  $selinux              = hiera("bdii::params::selinux",true)
-  $bdiipasswd           = hiera("bdii::params::bdiipasswd",randompass())
-  $bdiihost             = hiera("bdii::params::bdiihost","somehost.example.com")
-  $bdiideletedelay      = hiera("bdii::params::deletedelay", "0")
-  $slapdthreads         = hiera("bdii::params::slapdthreads", "16")
-  $slapdloglevel        = hiera("bdii::params::slapdloglevel", "0")
-
+  $log_level            = hiera('bdii::params::log_level','DEBUG')
+  $port                 = hiera('bdii::params::port','2170')
+  $user                 = hiera('bdii::params::user','ldap')
+  $slapdconf            = hiera('bdii::params::slapdconf','/etc/bdii/bdii-slapd.conf')
+  $selinux              = hiera('bdii::params::selinux','True')
```

Comment:
you're setting thee string to 'True' instead of using a bolean (still better than using the old unquoted True though...